pkg/etcdmgr/bootstrap/discovery.go

```diff
@@ -78,10 +84,9 @@ func queryDiscoveryService(token string) (*store.Event, error) {
 			if resp.StatusCode == http.StatusOK {
 				break
 			}
-			resp.Body.Close()
```

Comment:
there is potentially a response body for each iteration of the for loop. If all iterations except the last one fail, perhaps there is no body to close until we're done with the loop, but that's why the close was in each iteration of the loop.